apps/opik-frontend/src/components/pages-shared/traces/FeedbackScoresEditor/FeedbackScoresEditor.tsx

```diff
@@ -136,16 +159,27 @@ const FeedbackScoresEditor = ({
     <div className={cn("flex flex-col px-6", className)}>
       {header}
       <div className="grid max-w-full grid-cols-[minmax(0,5fr)_minmax(0,5fr)__36px_30px] border-b border-border empty:border-transparent">
-        {rows.map((row) => (
-          <AnnotateRow
-            key={row.name}
-            name={row.name}
-            feedbackDefinition={row.feedbackDefinition}
-            feedbackScore={row.feedbackScore}
-            onUpdateFeedbackScore={onUpdateFeedbackScore}
-            onDeleteFeedbackScore={onDeleteFeedbackScore}
-          />
-        ))}
+        {rows.map((row) => {
+          // Extract span_id from feedback score if it's a span feedback score
+          const spanIdForDelete = isSpanFeedbackScores
+            ? extractSpanMetadataFromValueByAuthor(
+                row.feedbackScore?.value_by_author,
+              ).span_id
+            : undefined;
+
+          return (
+            <AnnotateRow
+              key={row.name}
+              name={row.name}
+              feedbackDefinition={row.feedbackDefinition}
+              feedbackScore={row.feedbackScore}
+              onUpdateFeedbackScore={onUpdateFeedbackScore}
+              onDeleteFeedbackScore={(name: string) => {
+                onDeleteFeedbackScore(name, undefined, spanIdForDelete);
```

Comment:
I think you don't need to pass `spanIdForDelete` here, you can do it in a place where you use `FeedbackScoresEditor.tsx` by defining different onDeleteFeedbackScore. There is no need to overcomplicate this component